src/org/jitsi/impl/neomedia/rtcp/RTCPReceiverFeedbackTermination.java

```diff
@@ -0,0 +1,399 @@
+    /**
+     * Makes <tt>RTCPRRPacket</tt>s using information in FMJ.
+     *
+     * @return A <tt>List</tt> of <tt>RTCPRRPacket</tt>s to inject into the
+     * <tt>MediaStream</tt>.
+     */
+    private RTCPRRPacket[] makeRRs(long senderSSRC)
+    {
+        RTCPReportBlock[] reportBlocks = makeReportBlocks();
+        if (ArrayUtils.isNullOrEmpty(reportBlocks))
+        {
+            return null;
+        }
+
+        int mod = reportBlocks.length % MAX_RTCP_REPORT_BLOCKS;
+        int div = reportBlocks.length / MAX_RTCP_REPORT_BLOCKS;
+
+        RTCPRRPacket[] rrs = new RTCPRRPacket[mod == 0 ? div : div + 1];
+
+        // Since a maximum of 31 reception report blocks will fit in an SR
+        // or RR packet, additional RR packets SHOULD be stacked after the
+        // initial SR or RR packet as needed to contain the reception
+        // reports for all sources heard during the interval since the last
+        // report.
+        if (reportBlocks.length > MAX_RTCP_REPORT_BLOCKS)
+        {
+            int rrIdx = 0;
+            for (int off = 0;
+                 off < reportBlocks.length; off += MAX_RTCP_REPORT_BLOCKS)
+            {
+                int blockCount = Math.min(
+                    reportBlocks.length - off, MAX_RTCP_REPORT_BLOCKS);
+
+                RTCPReportBlock[] blocks = new RTCPReportBlock[blockCount];
+
+                System.arraycopy(reportBlocks, off, blocks, 0, blocks.length);
+
+                rrs[rrIdx++] = new RTCPRRPacket((int) senderSSRC, blocks);
+            }
+        }
+        else
+        {
+            rrs[0] = new RTCPRRPacket((int) senderSSRC, reportBlocks);
+        }
+
+        return rrs;
+    }
+
+    /**
+     * Iterate through all the <tt>ReceiveStream</tt>s that this
+     * <tt>MediaStream</tt> has and make <tt>RTCPReportBlock</tt>s for all of
+     * them.
+     *
+     * @return
+     */
+    private RTCPReportBlock[] makeReportBlocks()
+    {
+        // State validation.
+        if (stream == null)
+        {
+            logger.warn("stream is null.");
+            return MIN_RTCP_REPORT_BLOCKS_ARRAY;
+        }
+
+        StreamRTPManager streamRTPManager = stream.getStreamRTPManager();
+        if (streamRTPManager == null)
+        {
+            logger.warn("streamRTPManager is null.");
+            return MIN_RTCP_REPORT_BLOCKS_ARRAY;
+        }
+
+
+
+        // XXX MediaStreamImpl's implementation of #getReceiveStreams() says
+        // that, unfortunately, it has been observed that sometimes there are
+        // valid ReceiveStreams in MediaStreamImpl which are not returned by
+        // FMJ's RTPManager. Since (1) MediaStreamImpl#getReceiveStreams() will
+        // include the results of StreamRTPManager#getReceiveStreams() and (2)
+        // we are going to check the results against SSRCCache, it should be
+        // relatively safe to rely on MediaStreamImpl's implementation.
+        Collection<ReceiveStream> receiveStreams = stream.getReceiveStreams();
```

Comment:
Whaaaaaat? We should definitely not rely on FMJs ReceiveStream-s for this. Unless I am missing something we should have none of these whatsoever in the bridge anyway (they are fully decoded streams)